userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -1656,24 +1699,59 @@ static int32_t scap_bpf_handle_tp_mask(struct scap_engine_handle engine, uint32_
 		return SCAP_SUCCESS;
 	}
 
-	interesting_tp_set new_tp_set = {0};
+	interesting_ppm_tp_set new_tp_set = {0};
 	new_tp_set.tp[tp] = 1;
 	return load_tracepoints(handle, &new_tp_set);
 }
 
-static int32_t scap_bpf_handle_event_mask(struct scap_engine_handle engine, uint32_t op, uint32_t ppm_sc)
+static int32_t scap_bpf_handle_ppm_sc_mask(struct scap_engine_handle engine, uint32_t op, uint32_t ppm_sc)
```

Comment:
You can see the jank I made in ringbuffer/ which has similar constraints and either recoil in horror or do something similar (tl;dr macros and `static inline` functions).

Looks like we only use the engine/handle for two things:
* `->curr_sc_set` which we can pass in directly
* as an argument to the engine-specific functions, where it could be an opaque pointer if need be

See how e.g. ringbuffer/devset.c is basically engine independent even though it's the deepest dependency. Ugh. Did I invent mixin classes in C? Sorry ;)

Reply:
> You can see the jank I made in ringbuffer/ which has similar constraints and either recoil in horror or do something similar (tl;dr macros and static inline functions).

Yep, i would avoid that, but i think it's the best option. It can be also refactored in a scap_driver_engine_utils internal library, only linked by those engines.